Gems/Multiplayer/Code/Tests/PrefabProcessingTests.cpp

```diff
@@ -247,4 +283,46 @@ namespace UnitTest
             EXPECT_TRUE(IsChildAfterParent(childOfChildName, childName, entityList));
         }
     }
+
+    TEST_F(PrefabProcessingTestFixture, NetworkPrefabProcessor_ProcessPrefabEntityHierarchy_NestedNetworkEntityIsFound)
+    {
+        using AzToolsFramework::Prefab::PrefabConversionUtils::PrefabProcessorContext;
+
+        const AZStd::string parentName = "static_parent";
+        const AZStd::string childName = "static_child";
+        const AZStd::string childOfChildName = "networked_childOfChild";
+
+        // Create test entities with the following hierarchy:
+        // static parent
+        // + static child
+        //   + networked child
+        AZStd::vector<AZ::Entity*> entities;
+        AZ::Entity* parent = entities.emplace_back(CreateSourceEntity(parentName.c_str(), false, AZ::Transform::CreateIdentity()));
+        AZ::Entity* child = entities.emplace_back(CreateSourceEntity(childName.c_str(), false, AZ::Transform::CreateIdentity(), parent));
+        entities.emplace_back(CreateSourceEntity(childOfChildName.c_str(), true, AZ::Transform::CreateIdentity(), child));
+
+        // Convert the entities into prefab. Note: This will transfer the ownership of AZ::Entity* into Prefab
+        AzToolsFramework::Prefab::PrefabDom prefabDom;
+        ConvertEntitiesToPrefab(entities, prefabDom, "test_entities_sorted/path");
+
+        // Add the prefab into the Prefab Processor Context
+        const AZStd::string prefabName = "testPrefab";
+        TestPrefabProcessorContext prefabProcessorContext{ AZ::Uuid::CreateRandom() };
+        AzToolsFramework::Prefab::PrefabConversionUtils::PrefabDocument prefabDocument(prefabName);
+        ASSERT_TRUE(prefabDocument.SetPrefabDom(AZStd::move(prefabDom)));
+        prefabProcessorContext.AddPrefab(AZStd::move(prefabDocument));
+
+        // Request NetworkPrefabProcessor and PrefabCatchmentProcessor to process the prefab
+        Multiplayer::NetworkPrefabProcessor processor;
+        processor.Process(prefabProcessorContext);
+        AzToolsFramework::Prefab::PrefabConversionUtils::PrefabCatchmentProcessor prefabCatchmentProcessor;
+        prefabCatchmentProcessor.Process(prefabProcessorContext);
+        EXPECT_TRUE(prefabProcessorContext.HasCompletedSuccessfully());
+
+        // Verify entities are ordered by parent/child hierarchy
+        const auto& processedObjects = prefabProcessorContext.GetProcessedObjects();
+
+        // Verify that the nested network entity was discovered and 2 processed objects (spawnable and network.spawnable) were created
```

Comment:
comment looks incorrect. You are not verifying that the network entity is present in the network spawnable. You can probably create helper functions  and share it with the other tests:

- Helper to verify that a non-networked entity is in a non-networked spawnable
- Helper to verify that a networked entity is in a networked spawnable.

Maybe both of them can be merged into a single helper (spawnable has an entity)